app/controllers/application_controller.rb

```diff
@@ -1,20 +1,54 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
-  def require_admin
-      # ... and return is correct here
-      # rubocop:disable Style/AndOr
-      head :unauthorized and return unless @current_user.admin?
-      # rubocop:enable Style/AndOr
+  before_action :set_current_user
+  before_action :access_control
+  before_action :check_primary_account
+
+  private
+
+  def deny_access
+    if request.xhr?
+      head :unauthorized
+    else
+      render file: 'public/401.html',
+             status: :unauthorized,
+             layout: false
+    end
   end
+
+  # rubocop:disable AbcSize
```

Comment:
I sometimes like to put comments explaining why I'm disabling a cop locally to a particular method.

Here, perhaps to explain why you don't wish to refactor to have smaller ABC size.